src/integrator_interface.jl

```diff
@@ -417,58 +417,87 @@ function getobserved(integrator::DEIntegrator)
 end
 
 function sym_to_index(sym, integrator::DEIntegrator)
-    if has_sys(integrator.f) && is_state_sym(integrator.f.sys, sym)
-        return state_sym_to_index(integrator.f.sys, sym)
-    else
-        return sym_to_index(sym, getsyms(integrator))
+    idx = variable_index(integrator, sym)
+    if idx === nothing
+        idx = findfirst(isequal(sym), keys(integrator.u))
     end
+    return idx
+end
+
+# SymbolicIndexingInterface
+function SymbolicIndexingInterface.is_variable(A::DEIntegrator, sym)
+    is_variable(A.f, sym)
+end
+
+function SymbolicIndexingInterface.variable_index(A::DEIntegrator, sym)
+    variable_index(A.f, sym)
+end
+
+function SymbolicIndexingInterface.variable_symbols(A::DEIntegrator)
+    variable_symbols(A.f)
+end
+
+function SymbolicIndexingInterface.is_parameter(A::DEIntegrator, sym)
+    is_parameter(A.f, sym)
+end
+
+function SymbolicIndexingInterface.parameter_index(A::DEIntegrator, sym)
+    parameter_index(A.f, sym)
+end
+
+function SymbolicIndexingInterface.is_independent_variable(A::DEIntegrator, sym)
+    is_independent_variable(A.f, sym)
+end
+
+function SymbolicIndexingInterface.is_observed(A::DEIntegrator, sym)
+    return !is_variable(A, sym) && !is_parameter(A, sym) && !is_independent_variable(A, sym) && symbolic_type(sym) == ScalarSymbolic()
```

Comment:
This doesn't seem ideal. It seems like there should possibly be an easier way to define all of these.

Reply:
that sounds great!